kubernetes/provider.go

```diff
@@ -234,7 +233,47 @@ func (k kubeClientsets) AggregatorClientset() (*aggregator.Clientset, error) {
 	return k.aggregatorClientset, nil
 }
 
+func inCluster() bool {
```

Comment:
For this to only yield `true` when indeed run with valid in-cluster configuration, we'd also have to check for the presence of `/var/run/secrets/kubernetes.io/serviceaccount`.

The `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT` are only the endpoint information. Valid credentials are also required for a successful client connection. In the case of in-cluster config, they are injected via that volume mount of a service account token. That might not be the case when no service account is specified and `automountServiceAccountToken` is false.

Some details over here https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod